integration/spark/vendor/iceberg/src/main/java/io/openlineage/spark/agent/vendor/iceberg/metrics/CommitReportsFacetBuilder.java

```diff
@@ -0,0 +1,136 @@
+/*
+/* Copyright 2018-2024 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.spark.agent.vendor.iceberg.metrics;
+
+import io.openlineage.spark.agent.facets.IcebergCommitMetrics;
+import io.openlineage.spark.agent.facets.IcebergCommitReportOutputDatasetFacet;
+import java.util.Collections;
+import java.util.Optional;
+import org.apache.iceberg.metrics.CommitMetricsResult;
+import org.apache.iceberg.metrics.CommitReport;
+
+/** This class is responsible for building the CommitReportFacet from CommitReports. */
+public class CommitReportsFacetBuilder {
+
+  public static IcebergCommitReportOutputDatasetFacet from(CommitReport commitReport) {
+    return new IcebergCommitReportOutputDatasetFacet(
+        commitReport.snapshotId(),
+        commitReport.sequenceNumber(),
+        commitReport.operation(),
+        getIcebergCommitMetrics(commitReport.commitMetrics()),
+        commitReport.metadata(),
+        Collections.emptyMap());
+  }
+
+  private static IcebergCommitMetrics getIcebergCommitMetrics(CommitMetricsResult metricsResult) {
+    IcebergCommitMetrics.IcebergCommitMetricsBuilder builder = IcebergCommitMetrics.builder();
+
+    if (metricsResult == null) {
+      return builder.build();
+    }
+
+    Optional.ofNullable(metricsResult.totalDuration())
```

Comment:
Pretty sure it's possible to avoid all of those...

Reply:
Would love to hear one ;)

The problem is: there is Iceberg interface with ~20 methods, each of the methods returns `Counter` which can be null or may have long value. We need to translate it to our class as we don't want to build facets based on Iceberg classes directly.